pylint/checkers/variables.py

```diff
@@ -358,12 +358,13 @@ def _assigned_locally(name_node):
     return any(a.name == name_node.name for a in assign_stmts)
 
 
-def _is_type_checking_import(node):
-    parent = node.parent
-    if not isinstance(parent, nodes.If):
-        return False
-    test = parent.test
-    return test.as_string() in TYPING_TYPE_CHECKS_GUARDS
+def _is_type_checking_import(node: Union[nodes.Import, nodes.ImportFrom]) -> bool:
+    """Check if an import node is guarded by a TYPE_CHECKS guard"""
+    for ancestor in node.node_ancestors():
+        if isinstance(ancestor, nodes.If):
+            if ancestor.test.as_string() in TYPING_TYPE_CHECKS_GUARDS:
+                return True
+    return False
```

Comment:
This looks like it should be detected as a possible any() by the "consider-any-or-all" checker 

```suggestion
    return any(
        a.test.as_string() in TYPING_TYPE_CHECKS_GUARDS
        for a in node.node_ancestors()
        if isinstance(a, nodes.If)
    )
```

Reply:
Are you sure you find the `any` more readable? In this case I personally prefer the `for` loop, but happy to commit if you prefer this one.

I've also stopped adding to `2.12` but thought that `--enable-all-extensions` would be valuable to try and squeeze in.

Reply:
Well it's kinda opinionated at this point, in fact I think it should be

```python
    return any(
        isinstance(a, nodes.If) and a.test.as_string() in TYPING_TYPE_CHECKS_GUARDS
        for a in node.node_ancestors()
    )
```   

But arguably it's very dense so readability is not great.

>  that --enable-all-extensions would be valuable to try and squeeze in.

Well, it did, just in time to be number 200 😉

Reply:
I'll leave this to you 😄

Reply:
Let's keep the for loop

Reply:
Btw, since I did not open an issue this wouldn't push the 200 limit 👀

Reply:
I'm actually more confident in a 200 issues/MR release with primer tests and typing than I was for 2.10 and 2.9 that had around 100 issues/MR btw 😄 We're reducing risk drastically !

Reply:
Certaintly! Especially after we fixed 3 (4?) crashes already identified by the primer. Our work payed off immediately!

Reply:
> we fixed 3 (4?) crashes

This did not feel like it, because it felt easy and small compared to the primer tests and configuration test functional framework, but this is huge ! The crash issues that we have to handle fast just after a release are the most stressful one. Looking forward to release being uneventful !

Reply:
It helped that I was stuck on an airplane twice last week 😄